File: API_MyLibrary/objects/type.php
<?php
    /*  Projet  : API_MyLibrary
        Auteur  : Svoboda Karel Vilém
        Date    : 03.05.2022
        Version : 0.1

        Class   : Type
        Desc.   : Permet de répliquer les données dans la table Types
    */

    class Type
    {
        //Variables d'instances
        private $_idType, $_nomType, $_nomImage, $_auteur, $_description;

        //Propriétés
        public function getIdType(){
            return $this->_idType;
        }
        public function setIdType($idType){
            $this->_idType = $idType;
        }

        public function getNomType(){
            return $this->_nomType;
        }
        public function setNomType($nomType){
            $this->_nomType = $nomType;
        }

        public function getNomImage(){
            return $this->_nomImage;
        }
        public function setNomImage($nomImage){
            $this->_nomImage = $nomImage;
        }

        public function getAuteur(){
            return $this->_auteur;
        }
        public function setAuteur($auteur){
            $this->_auteur = $auteur;
        }

        public function getDescription(){
            return $this->_description;
        }
        public function setDescription($description){
            $this->_description = $description;
        }

        

        //Constructeur
        /** Permet de créer un Type de reference
         * 
         *  @param integer int(11) $idType
         *  @param string varchar(255) $nomType
         *  @param string varchar(255) $nomImage
         *  @param string varchar(100) $auteur
         *  @param string mediumtext $description
         */
        function __construct($idType, $nomType, $nomImage, $auteur, $description)
        {
            $this->_idType = $idType;
            $this->_nomType = $nomType;
            $this->_auteur = $auteur;
            $this->_nomImage = $nomImage;
            $this->_description = $description;
        }
    }
?>

File: src/Projet C#/MyLibrary/WindowsFormsApp1/classes/tablesBaseDeDonnees/Type.cs
/* Projet   : MyLibrary - TPI 2022
 * Version  : 0.8.1
 * Date     : 16.05.2022
 * 
 * Auteur   : Karel V. Svoboda
 * Classe   : I.DA-P4A
 * 
 * Class    : Type.cs class
 * Decs.    : Permet d'encapsuler la table Types de la base de données
 */

namespace MyLibrary.classes
{
    public class Type
    {
        #region Variables d'instances
        private int _idType;
        private string _nomType;
        #endregion

        #region Propriétés
        public int IdType
        {
            get { return _idType; }
            set { _idType = value; }
        }

        public string NomType
        {
            get { return _nomType; }
            set { _nomType = value; }
        }
        #endregion

        #region Constructeurs
        /// <summary>
        /// Permet d'encapsuler la table Types de la base de données
        /// </summary>
        /// <param name="idType">int(11)</param>
        /// <param name="nomType">varchar(255)</param>
        public Type(int idType, string nomType)
        {
            _idType = idType;
            _nomType = nomType;
        }
        #endregion
    }
}
